src/Utils/Hooks/useAuthValidation.ts

```diff
@@ -17,7 +17,7 @@ export const useAuthValidation = () => {
           }
         `,
         {}
-      ).toPromise()
+      )?.toPromise()
```

Comment:
Resolves `TypeError: Cannot read properties of undefined (reading 'toPromise')`.

Reply:
I just tested this again, running node v18.15.0 and latest version of Jest.

As suggested I ran `yarn relay` tho there were no changes to generated files.

<details>

<summary>Full trace</summary>

```bash
➜  yarn jest src/Apps/Components/__tests__/AppShell.jest.tsx
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
/Users/lalalala/code/force/src/Utils/Hooks/useAuthValidation.ts:37
                return (0, _reactRelay.fetchQuery)(relayEnvironment, _useAuthValidationQuery !== void 0 ? _useAuthValidationQuery : (_useAuthValidationQuery = require("../../__generated__/useAuthValidationQuery.graphql"), _useAuthValidationQuery.hash && _useAuthValidationQuery.hash !== "2f2ccc765f2b0623e0c21ddb7e46eb08" && console.error("The definition of 'useAuthValidationQuery' appears to have changed. Run `relay-compiler` to update the generated files to receive the expected data."), _useAuthValidationQuery), {}).toPromise();
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         ^

TypeError: Cannot read properties of undefined (reading 'toPromise')
    at _callee$ (/Users/lalalala/code/force/src/Utils/Hooks/useAuthValidation.ts:37:522)
    at tryCatch (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:274:22)
    at Generator.prototype.<computed> [as next] (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at /Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
    at new Promise (<anonymous>)
    at /Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12
    at exec (/Users/lalalala/code/force/src/Utils/Hooks/useAuthValidation.ts:73:21)
    at /Users/lalalala/code/force/src/Utils/Hooks/useAuthValidation.ts:78:5
    at invokePassiveEffectCreate (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:23487:20)
    at HTMLUnknownElement.callCallback (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:3945:14)
    at HTMLUnknownElement.callTheUserObjectsOperation (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
    at innerInvokeEventListeners (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:350:25)
    at invokeEventListeners (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:286:3)
    at HTMLUnknownElementImpl._dispatch (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:233:9)
    at HTMLUnknownElementImpl.dispatchEvent (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:104:17)
    at HTMLUnknownElement.dispatchEvent (/Users/lalalala/code/force/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
    at Object.invokeGuardedCallbackDev (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:3994:16)
    at invokeGuardedCallback (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:4056:31)
    at flushPassiveEffectsImpl (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:23574:9)
    at unstable_runWithPriority (/Users/lalalala/code/force/node_modules/scheduler/cjs/scheduler.development.js:468:12)
    at runWithPriority$1 (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:11276:10)
    at flushPassiveEffects (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:23447:14)
    at performSyncWorkOnRoot (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:22269:3)
    at /Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:11327:26
    at unstable_runWithPriority (/Users/lalalala/code/force/node_modules/scheduler/cjs/scheduler.development.js:468:12)
    at runWithPriority$1 (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:11276:10)
    at flushSyncCallbackQueueImpl (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:11322:9)
    at flushSyncCallbackQueue (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:11309:3)
    at batchedUpdates$1 (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom.development.js:22387:7)
    at act (/Users/lalalala/code/force/node_modules/react-dom/cjs/react-dom-test-utils.development.js:1042:14)
    at render (/Users/lalalala/code/force/node_modules/@testing-library/react/dist/pure.js:97:26)
    at _callee$ (/Users/lalalala/code/force/src/Apps/Components/__tests__/AppShell.jest.tsx:46:5)
    at tryCatch (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:274:22)
    at Generator.prototype.<computed> [as next] (/Users/lalalala/code/force/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/Users/lalalala/code/force/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.15.0
 FAIL  src/Apps/Components/__tests__/AppShell.jest.tsx
  ● Test suite failed to run

    Jest worker encountered 4 child process exceptions, exceeding retry limit

      at ChildProcessWorker.initialize (node_modules/jest-cli/node_modules/jest-worker/build/workers/ChildProcessWorker.js:181:21)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        25.061 s
Ran all test suites matching /src\/Apps\/Components\/__tests__\/AppShell.jest.tsx/i.
error Command failed with exit code 1.
```

</details>

Test passed with the change discussed here.